### PR TITLE
Allow fullscreen by default.

### PR DESCRIPTION
**Pull request by adityaruplaha**
Follow up of #3570 .
# Purpose
* Fix the stuck-in-fullscreen glitch 
* Add a QCheckBox for starting in fullscreen by default.

# Major Design Changes
* `EmuWindow_SDL2` now takes a boolean indicating whether to use fullscreen into its constructor. (in #3607)
* The fullscreen checkbox in the Qt frontend is now disabled when emulation is not running. (Upon launch, bootgame, gameclose, the value is taken from `Settings::values.default_fullscreen`)
>So, if the users unchecks fullscreen, closes the game while default_fullscreen is enabled, the next game 
will also start in full screen, automatically. This is intended behaviour.

---
I've split out some changes to #3607 

# Screenshot:
capture

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3576)

### CODE REVIEW

File: src/citra/citra.cpp [hunk 1]
```diff
@@ -210,6 +217,10 @@ int main(int argc, char** argv) {
             case 'p':
                 movie_play = optarg;
                 break;
+            case 'f':
+                fullscreen = true;
+                std::cout << "Starting in fullscreen mode...\n";
```
Comment (daniellimws) [inline src/citra/citra.cpp]: You might wanna use the logging macros for this

`NGLOG_INFO(Frontend, "Starting in fullscreen mode...")`

Comment (adityaruplaha) [reply src/citra/citra.cpp]: Will do.

File: src/citra_qt/main.cpp [hunk 2]
```diff
@@ -1163,9 +1165,13 @@ void GMainWindow::closeEvent(QCloseEvent* event) {
         return;
     }
 
-    UISettings::values.geometry = saveGeometry();
+    // Don't save geometry if fullscreen was on
+    if (!ui.action_Fullscreen)
```
Comment (FearlessTobi) [inline src/citra_qt/main.cpp]: Why do you use two seperate if statements for the same check?

**Comment by FearlessTobi** [comment]
When I tick the checkbox and restart Citra, it still doesn't start in fullscreen.

**Comment by FearlessTobi** [comment]
When I tick the checkbox, restart Citra, start a game, go out of fullscreen and start another game, Citra doesn't put the game in fullscreen mode. Do you want this behavior for some reason or this a bug?

**Comment by adityaruplaha** [comment]
It is normal (you went out of full-screen). But I will change it tomorrow, I realised it just now.

File: src/citra_qt/main.cpp (hunk 2 above)
Comment (adityaruplaha) [reply src/citra_qt/main.cpp]: They are not the same check. One is for Single Window mode & another is for rendering in seperate window.

Comment (adityaruplaha) [reply src/citra_qt/main.cpp]: Oh, I'm an idiot. Will fix.

Comment (adityaruplaha) [reply src/citra_qt/main.cpp]: Fixed.

File: src/citra/citra.cpp (hunk 1 above)
Comment (adityaruplaha) [reply src/citra/citra.cpp]: Done.

**Comment by adityaruplaha** [comment]
Ok, I've disabled the checkbox (you can't change it) when emulation is not running, similarly to the F11 toggle being disabled.

From Discord:
capture

File: src/citra/emu_window/emu_window_sdl2.h [hunk 3]
```diff
@@ -12,7 +12,7 @@ struct SDL_Window;
 
 class EmuWindow_SDL2 : public EmuWindow {
 public:
-    EmuWindow_SDL2();
+    EmuWindow_SDL2(bool fullscreen);
```
[the remark this answers is not in the document]
Comment (adityaruplaha) [reply src/citra/emu_window/emu_window_sdl2.h]: Ok, will fix.

**Comment by adityaruplaha** [comment]
But why does `fullscreen` have to be removed? Because it's state isn't saved anymore? In that case, will fix.

File: src/citra_qt/main.cpp [hunk 4]
```diff
@@ -904,6 +913,7 @@ void GMainWindow::OnPauseGame() {
 }
 
 void GMainWindow::OnStopGame() {
+
```
Comment (FearlessTobi) [inline src/citra_qt/main.cpp]: Is this change intended?

**Comment by jroweboy** [comment]
* we don't need another setting option for this. ask yourself how many games have two options for fullscreen? its really unclear what each is supposed to do. roll the logic into the already existing fullscreen option.
* you already had a pr for this. in the future, don't open a new pr, just update the old pr and reopen it.

**Comment by adityaruplaha** [comment]
I couldn't see the reopen option, for since reason. And I'm pretty sure that quite a few programs have a "full-screen by default" option (Eg.: AoE II, AoE III, Star Wars KoToR, CS). What they don't have is a checkbox fir fullscreen. How many users use it anyway, provided that a hotkey exists?

Now I'm suggesting a design which will be simpler for most users. I don't care if users don't even bother to check the hotkeys. I'm okay with keeping the existing checkbox, but believe me, this design is simpler. (I've already done some intuition checks.)

File: src/citra_qt/main.cpp (hunk 4 above)
Comment (adityaruplaha) [reply src/citra_qt/main.cpp]: Clang format

**Comment by jroweboy** [comment]
> "full-screen by default" option (Eg.: AoE II, AoE III, Star Wars KoToR, CS)

You listed games that have a SINGLE fullscreen option! Where are you getting that having a fullscreen in the view menu AND a separate "fullscreen by default" in the config menu idea from? heres AoE2 https://i.imgur.com/oW7TNbI.png and heres AoE3 https://i.imgur.com/AvH3bis.png Apologies that i don't have the other games to show you screenshots from their settings, but i'm guessing they also have a single "fullscreen option". For a more recent example: heres overwatch https://i.imgur.com/B4fJ0ja.png Once again, a single option that just works.

> Now I'm suggesting a design which will be simpler for most users

No. No no no, two settings that deal with fullscreen is NOT easier to understand than a fullscreen that "just works". Please don't try to pull this crap on me; its really trying my patience.

Heres how simple fullscreen should be:

Fullscreen does not make citra-qt go fullscreen unless its in game. Get rid of this silly fullscreen citra-qt ui, until someone makes a fullscreen friendly ui.

All the fullscreen option (the ones thats already there in the settings and in the View menu) should do is make the render window go fullscreen when a game launches and fullscreens it on the current monitor. Then if someone launches a game from the command line (ie through steam), it doesn't matter, the game is fullscreen. It also removes the silly issue with having a fullscreen UI window that was reported before where you can't unfullscreen citra ui if you closed it when it was last fullscreened. 

With the changes you have, you can pop in and out of fullscreen all you want in game with alt-enter, but the fullscreen mode isn't persisted between citra relaunches without hitting that hidden checkbox. Thats where the issue is, if I closed my game in fullscreen last time, I expect next time i launch the game to be in fullscreen. If i close the game not in fullscreen, I expect it to not be in fullscreen next time i launch it. A second option is a complete waste.

Good design isn't adding more checkboxes for supporting every last possible configuration people could theoretically ask for.

**Comment by adityaruplaha** [comment]
> If i close the game not in fullscreen, I expect it to not be in fullscreen next time i launch it.

Precisely the opposite of what @fearlessTobi told me would be intuitive. Now what do I do? _feeling like jumping off the balcony_

**Comment by adityaruplaha** [comment]
Ok, conflicts resolved, will complete this soon.

File: src/citra/emu_window/emu_window_sdl2.cpp [hunk 5]
```diff
@@ -81,12 +81,26 @@ EmuWindow_SDL2::EmuWindow_SDL2() {
...
     std::string window_title = Common::StringFromFormat("Citra %s| %s-%s ", Common::g_build_name,
                                                         Common::g_scm_branch, Common::g_scm_desc);
-    render_window =
-        SDL_CreateWindow(window_title.c_str(),
-                         SDL_WINDOWPOS_UNDEFINED, // x position
-                         SDL_WINDOWPOS_UNDEFINED, // y position
-                         Core::kScreenTopWidth, Core::kScreenTopHeight + Core::kScreenBottomHeight,
-                         SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE | SDL_WINDOW_ALLOW_HIGHDPI);
+
+    if (fullscreen) {
```
Comment (VPeruS) [inline src/citra/emu_window/emu_window_sdl2.cpp]: Consider to remove SDL_CreateWindow out of if statement

Comment (adityaruplaha) [reply src/citra/emu_window/emu_window_sdl2.cpp]: Sorry, that's really impossible, unless you are asking me to modify the width & height later on? Well, it doesn't make sense to create a window using some values & changing then immediately, does it? I think it's computationally expensive too.

**Comment by adityaruplaha** [comment]
I'm still cleaning up the UI for the default full screen feature, so will it be ok if I do the SDL2 changes in another PR (coz that'll be the majority use case) & rebase this off master when it's merged?

File: src/citra/emu_window/emu_window_sdl2.cpp (hunk 5 above)
Comment (adityaruplaha) [reply src/citra/emu_window/emu_window_sdl2.cpp]: Never mind. I'm rewriting this code soon.

Comment (VPeruS) [reply src/citra/emu_window/emu_window_sdl2.cpp]: I mean to calculate size in `if` and pass it to SDL_CreateWindow

File: src/citra/emu_window/emu_window_sdl2.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (adityaruplaha) [reply src/citra/emu_window/emu_window_sdl2.h]: Fixed it in #3607

File: src/citra/emu_window/emu_window_sdl2.cpp (hunk 5 above)
Comment (adityaruplaha) [reply src/citra/emu_window/emu_window_sdl2.cpp]: Rewritten in #3607

**Comment by adityaruplaha** [comment]
@MerryMage I'm not removing that variable right now as it might come in handy in the future (of this PR).

**Comment by adityaruplaha** [comment]
I think this PR is unnecessary as the full-screen state gets restored correctly by #3611. Closing...
